atomate/vasp/workflows/base/exchange.py

```diff
@@ -0,0 +1,353 @@
+# coding: utf-8
+
+import os
+import numpy as np
+
+from fireworks import Workflow, Firework
+from atomate.vasp.fireworks.core import StaticFW
+from atomate.vasp.powerups import (
+    add_tags,
+    add_modify_incar,
+    add_additional_fields_to_taskdocs,
+    add_wf_metadata,
+    add_common_powerups,
+)
+from atomate.vasp.workflows.base.core import get_wf
+from atomate.vasp.fireworks.exchange import HeisenbergModelFW, VampireCallerFW
+from atomate.vasp.firetasks.exchange_tasks import ExchangeToDB
+from atomate.vasp.config import VASP_CMD, DB_FILE, ADD_WF_METADATA
+from atomate.vasp.workflows.presets.scan import wf_scan_opt
+from atomate.utils.utils import get_logger
+
+from uuid import uuid4
+from copy import deepcopy
+
+logger = get_logger(__name__)
+
+from pymatgen.io.vasp.sets import MPStaticSet
+from pymatgen.analysis.structure_matcher import StructureMatcher, ElementComparator
+from pymatgen.analysis.magnetism.analyzer import (
+    CollinearMagneticStructureAnalyzer,
+    MagneticStructureEnumerator,
+)
+
+__author__ = "Nathan C. Frey"
+__maintainer__ = "Nathan C. Frey"
+__email__ = "ncfrey@lbl.gov"
+__status__ = "Development"
+__date__ = "July 2019"
+__version__ = 0.1
+
+
+class ExchangeWF:
```

Comment:
It seems like this overlaps considerably with `MagneticOrderingsWF`. Could this be simplified by adding an option to only perform static calculations in `MagneticOrderingsWF` and then use that internally to generate the most of the fireworks when `static=True`.

@mkhorton  I don't know if you have any thoughts on this?

Reply:
Yes, major overlap. The `MagneticOrderingsWF` could be modified to offer a static-only option, and also a "static-only with constrained noncollinear moments" option.

Reply:
Ok. If it's not too much trouble to at least add the static only option I think that would help reduce the complexity of this pull request a lot.

Reply:
Hi @ncfrey. I'm a bit confused about what's happened to this workflow, it seems like you remove the static option?

Reply:
Hi @utf, after discussing with @mkhorton it seemed like it would be easiest to kind of partition all magnetic ordering calculations into the `MagneticOrderingsWF`, so I moved the static only option there. 

We are going to discuss how that can be extended/modified to include the more robust options I've mentioned.

My idea was that now the `ExchangeWF` is totally agnostic to how the magnetic structures and energies are generated, which was a goal I had at the beginning. Let me know if anything is unclear.

Reply:
That makes sense! Thanks for clearing that up.